diff --git a/dlls/jscript/Makefile.in b/dlls/jscript/Makefile.in
index 0919cfe9bf031b4788c55ebd4b66bd6f35e61514..ef1a809951ccb94f1b89534a5e07b4ae083d93e0 100644
--- a/dlls/jscript/Makefile.in
+++ b/dlls/jscript/Makefile.in
@@ -1,6 +1,9 @@
 MODULE    = jscript.dll
 IMPORTS   = oleaut32 ole32 user32 advapi32
 
+VER_FILEDESCRIPTION_STR = "Wine JScript"
+VER_PRODUCTVERSION      = 5,8,7601,16982
+
 SOURCES = \
 	activex.c \
 	array.c \
diff --git a/dlls/jscript/jscript.rc b/dlls/jscript/jscript.rc
index cbddcb90058247cb4cd5c24827b50ddc87ac4938..8b6f78818bd9897b5bcf324cf46811a2d7fa7ded 100644
--- a/dlls/jscript/jscript.rc
+++ b/dlls/jscript/jscript.rc
@@ -102,12 +102,3 @@ LANGUAGE LANG_NEUTRAL, SUBLANG_NEUTRAL
 
 /* @makedep: jscript.rgs */
 2 WINE_REGISTRY jscript.rgs
-
-#define WINE_FILEDESCRIPTION_STR "Wine JScript"
-#define WINE_FILENAME_STR "jscript.dll"
-#define WINE_FILEVERSION 5,7,7601,16982
-#define WINE_FILEVERSION_STR "5.8.7601.16982"
-#define WINE_PRODUCTVERSION 5,8,7601,16982
-#define WINE_PRODUCTVERSION_STR "5.8.7601.16982"
-
-#include "wine/wine_common_ver.rc"
diff --git a/dlls/mshtml/Makefile.in b/dlls/mshtml/Makefile.in
index bc712a202c5a2230fa95c5f7823539650c8c676e..6cce21037ae67697c0dde7ae1fc8f829052224e5 100644
--- a/dlls/mshtml/Makefile.in
+++ b/dlls/mshtml/Makefile.in
@@ -3,6 +3,10 @@ IMPORTLIB = mshtml
 IMPORTS   = strmiids uuid urlmon shlwapi shell32 ole32 oleaut32 user32 gdi32 advapi32
 DELAYIMPORTS = wininet inetcomm
 
+VER_FILEDESCRIPTION_STR = "Wine HTML Viewer"
+VER_PRODUCTVERSION      = 6,0,2800,1106
+VER_OLESELFREGISTER     = 1
+
 SOURCES = \
 	conpoint.c \
 	dispex.c \
diff --git a/dlls/mshtml/mshtml.rc b/dlls/mshtml/mshtml.rc
index 11083f29da58e5cabc9be0eeb5d243d97cace1b4..398ac550ccb4ca9f0f9e80c8e6e5b64ebd440732 100644
--- a/dlls/mshtml/mshtml.rc
+++ b/dlls/mshtml/mshtml.rc
@@ -59,16 +59,6 @@ FONT 8, "MS Shell Dlg"
 
 LANGUAGE LANG_NEUTRAL, SUBLANG_NEUTRAL
 
-#define WINE_FILEDESCRIPTION_STR "Wine HTML Viewer"
-#define WINE_FILENAME_STR "mshtml.dll"
-#define WINE_FILEVERSION 6,0,2800,1106
-#define WINE_FILEVERSION_STR "6.0.2800.1106"
-#define WINE_PRODUCTVERSION 6,0,2800,1106
-#define WINE_PRODUCTVERSION_STR "6.0.2800.1106"
-#define WINE_EXTRAVALUES VALUE "OLESelfRegister",""
-
-#include "wine/wine_common_ver.rc"
-
 /* @makedep: mshtml.inf */
 REGINST REGINST mshtml.inf
 

