diff --git a/include/wine/test.h b/include/wine/test.h
index 8f3f3bb0d50..5085c442fed 100644
--- a/include/wine/test.h
+++ b/include/wine/test.h
@@ -35,6 +35,28 @@
 #define INVALID_SET_FILE_POINTER (~0u)
 #endif

+#ifdef NONAMELESSUNION
+# define U(x)  (x).u
+# define U1(x) (x).u1
+# define U2(x) (x).u2
+# define U3(x) (x).u3
+# define U4(x) (x).u4
+# define U5(x) (x).u5
+# define U6(x) (x).u6
+# define U7(x) (x).u7
+# define U8(x) (x).u8
+#else
+# define U(x)  (x)
+# define U1(x) (x)
+# define U2(x) (x)
+# define U3(x) (x)
+# define U4(x) (x)
+# define U5(x) (x)
+# define U6(x) (x)
+# define U7(x) (x)
+# define U8(x) (x)
+#endif
+
 /* debug level */
 extern int winetest_debug;

